dapp/src/components/dialogs/GeneralInfoDialog.tsx

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+'use client';
+
+import {
+    Dialog,
+    DialogBody,
+    DialogContent,
+    DialogPosition,
+    Header,
+    KeyValueInfo,
+    truncate,
+} from '@iota/apps-ui-kit';
+import { useCurrentAccount } from '@iota/dapp-kit';
+import Link from 'next/link';
+
+import { useNameRecord, useRegistrationNfts } from '@/hooks';
+import { formatDate } from '@/lib/utils/format/formatDate';
+import { normalizeNameInput } from '@/lib/utils/format/formatNames';
+
+import { CollapsibleInfo } from '../CollapsibleInfo';
+import { AvatarDisplay } from '../name-record/AvatarDisplay';
+
+type UpdateNameDialogProps = {
+    name: string;
+    open: boolean;
+    setOpen: (bool: boolean) => void;
+};
+
+export function GeneralInfoDialog({ name, open, setOpen }: UpdateNameDialogProps) {
+    const account = useCurrentAccount();
+    const address = account?.address || '';
+
+    const { data: names } = useRegistrationNfts('name');
+    const { data: subnames } = useRegistrationNfts('subname');
+
+    const isSubname = !names?.find((n) => n.name === name);
+    const currentName = isSubname
+        ? subnames?.find((n) => n.name === name)
+        : names?.find((n) => n.name === name);
+
+    const currentNameId = currentName?.id ?? '';
+    const expirationTime = currentName?.expirationTimestampMs;
+
+    const { data: nameRecordData } = useNameRecord(name);
+    const targetAddress =
+        nameRecordData?.type === 'unavailable'
+            ? nameRecordData.nameRecord.targetAddress
+            : undefined;
+
+    const handleClose = () => setOpen(false);
+    type InfoLinks = { key: string; value: string; href: string };
+    const InfoLinks: InfoLinks[] = [
+        {
+            key: 'Owner',
+            value: address,
+            href: `https://explorer.iota.org/address/${address}`,
+        },
+        targetAddress && {
+            key: 'Target address',
+            value: targetAddress,
+            href: `https://explorer.iota.org/address/${targetAddress}`,
+        },
+        {
+            key: 'Object ID',
+            value: currentNameId,
+            href: `https://explorer.iota.org/address/${currentNameId}`,
```

Comment:
so that it opens directly on devnet. We can improve the link generation in another PR

```suggestion
            href: `https://explorer.iota.org/address/${address}?network=devnet`,
        },
        targetAddress && {
            key: 'Target address',
            value: targetAddress,
            href: `https://explorer.iota.org/address/${targetAddress}?network=devnet`,
        },
        {
            key: 'Object ID',
            value: currentNameId,
            href: `https://explorer.iota.org/address/${currentNameId}?network=devnet`,
```